custom_components/elasticsearch/es_doc_creator.py

```diff
@@ -157,33 +157,53 @@ def _state_to_entity_details(self, state: State) -> dict:
         """
         entity_details = self._entity_details.async_get(state.entity_id)
 
-        additions = {}
-
-        if entity_details is not None:
-            additions["device"] = {}
-
-            if entity_details.device:
-                additions["device"]["id"] = entity_details.device.id
-                additions["device"]["name"] = entity_details.device.name
+        if entity_details is None:
+            return {}
+
+        entity = entity_details.entity
+        entity_capabilities = entity.capabilities or {}
+        entity_area = entity_details.entity_area
+
+        entity_additions = {
+            "labels": entity_details.entity_label,
+            "name": entity.name or entity.original_name,
+            "friendly_name": state.name,
+            "platform": entity.platform,
+            "unit_of_measurement": str(entity.unit_of_measurement),
+            "area.id": entity_area.id if entity_area else None,
+            "area.name": entity_area.name if entity_area else None,
+            "class": entity_capabilities.get("state_class"),
+        }
 
-            if entity_details.entity_area:
-                additions["area"] = {
-                    "id": entity_details.entity_area.id,
-                    "name": entity_details.entity_area.name,
-                }
+        entity_additions = {
+            k: str(v)
+            for k, v in entity_additions.items()
+            if (v is not None and v != "None")
+        }
 
-            if entity_details.entity.platform:
-                additions["platform"] = entity_details.entity.platform
-            if entity_details.entity.name:
-                additions["name"] = entity_details.entity.name
+        device = entity_details.device
+        device_floor = entity_details.device_floor or {}
+        device_area = entity_details.device_area
+
+        device_additions = {
+            "class": entity.device_class or entity.original_device_class,
+            "id": (device.id if device else None),
+            "labels": entity_details.device_label,
+            "name": (device.name if device else None),
+            "friendly_name": (device.name_by_user if device else None),
+            "floor.id": device_floor.get("id"),
```

Comment:
Same comment for `floor.*` and `area.*` with respect to flattening the properties.

Reply:
Commented above